src/Routing/Route/Route.php

```diff
@@ -423,6 +423,7 @@ public function match(array $url, array $context = [])
             $hostOptions['_ext'] = $url['_ext'];
             unset($url['_ext']);
         }
+        unset($defaults['_ext']);
```

Comment:
If I understood your question correctly, it would break parsing. I tried doing the change in `Cake\Routing\RouteBuilder::resources()` instead but parsing broke.